.github/workflows/check-size.yml

```diff
@@ -0,0 +1,22 @@
+name: Check binary size
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled, unlabeled]
+
+jobs:
+  check_size:
+    runs-on: ubuntu-latest
+    # Check if pull request does not have label "bloat-ok".
+    if: "!contains(github.event.pull_request.labels.*.name, 'bloat-ok')"
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        # Use HEAD from the PR instead of the merge commit
+        ref: ${{ github.event.pull_request.head.sha }}
+        # Needed to rebase against the base branch
+        fetch-depth: 0
+    - name: Configure base branch without switching current branch
+      run: git fetch origin ${GITHUB_BASE_REF}:${GITHUB_BASE_REF}
+    - name: Compare binary size change across each commit
+      run: git rebase ${GITHUB_BASE_REF}^ -x test/check-size.sh
```

Comment:
Do we need to run `check-size.sh` here twice -- with and without `git rebase ${GITHUB_BASE_REF}^`?

Reply:
I'm not sure I understand. The `rebase -x` lets us detect which commit is the offender, whereas running against HEAD without that will only give us a yes/no for the bloating.

Reply:
The first run of `check-size.sh` it would create `prev_size` with the size of the binary from the _main_ branch.
The second run would build a binary with commits from the pull request and compare the size.

Reply:
Ah, that would require a checkout to main, and back to the PR. Instead, doing `rebase main^` with the `^` lets us use the parent commit from main as the first one in the rebase process.

Reply:
For example, running the following commit executes only the `else` statement within the script:
```
$ git rebase main^ -x test/check-size.sh
Executing: test/check-size.sh
go build -buildmode=pie -o checkpointctl -ldflags "-X main.name=checkpointctl -X main.version=0.1.0"
No previous size present, storing current size
Successfully rebased and updated refs/heads/check-bin-size.
```

Reply:
:thinking: I'm not sure how you managed to make it run just once. When I tried running `git rebase main^ -x test/check-size.sh` on a branch one commit ahead of `check-bin-size`, the script ran twice - once with just the `else` part, and a second time for the new commit to compare the size. I borrowed the command itself straight from podman, so I'm quite sure it covers all commits, including the latest one. Still not understanding why we would have to run the script by itself once again :/

Reply:
Ah, you are right. I ran the command on the `check-bin-size` branch while having the same commits applied to the main branch. Since the head commits were the same `git rebase` executed the script only once.

I was tested again on another branch with different commits and it works as expected. Thanks for pointing this out.